cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/instantsearch-config.js

```diff
@@ -576,8 +573,22 @@ function enableInstantSearch(config) {
      * @return {any} A color swatch
      */
     function renderSwatches(hit, html) {
+        let colorFacets = [];
+        let sizeFacets = [];
+        if (algoliaResults && algoliaResults._state) {
```

Comment:
Using global variables should be a last resort, it's more prone to bugs as you could see on the promotional pricing PR.
You have the `hit` in the function, so you know what is the currently displayed variant. Why don't you just rely on the `color` of the product and select the matching swatch?

For the product-level model, you can just add the selectedColor to the hit in the `transformItem` function.